File: Exercism/fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

let toRna (dna: string): string =
    dna
    |> String.map
        (fun c ->
            match c with
            | 'C' -> 'G'
            | 'G' -> 'C'
            | 'T' -> 'A'
            | 'A' -> 'U'
            | _ -> c)

File: Exercism/csharp/rna-transcription/RnaTranscription.cs
using System;
using System.Linq;

public static class RnaTranscription
{
    private static char Match(char nucleotide) => nucleotide switch { 'G' => 'C', 'C' => 'G', 'T' => 'A', 'A' => 'U', var n => throw new ArgumentException($"Invalid nucleotide: {n}") };

    public static string ToRna(string nucleotide) => string.Concat(nucleotide.Select(Match));
}